lib/spack/spack/fetch_strategy.py

```diff
@@ -712,6 +712,11 @@ def fetch(self):
     def archive(self, destination):
         super(GitFetchStrategy, self).archive(destination, exclude='.git')
 
+    def get_commit_hash(self):
+        git_exe = self.git.name
+        git_commit = subprocess.check_output([git_exe, 'rev-parse', 'HEAD']).split()
```

Comment:
I have to look into this a bit more, but spack has this `command` stuff. No need for `subprocess`:

https://github.com/spack/spack/blob/4ecb465a366de3f722aeb1525b22c0d22aefceae/var/spack/repos/builtin/packages/spark/package.py#L51-L53

`check_output` is not available for earlier Python versions.

Reply:
I was trying to do that but I failed in the past. Now it works fine.
Thanks for pointing that out :+1: